jit-service/src/main/java/org/dcsa/jit/service/TimestampRoutingService.java

```diff
@@ -4,44 +4,142 @@
   private final OutboxMessageRepository outboxMessageRepository;
   private final ObjectMapper objectMapper;
 
+  @Override
+  public void configure() throws Exception {
+    pollOutstandingRoutingMessages();
+    emitMessage();
+    tokenOIDC();
+    timestampNotification();
+  }
+
+  void pollOutstandingRoutingMessages() {
+    from("{{camel.route.outbox-msg}}")
+        .routeId("outbox-message-poll")
+        .errorHandler(noErrorHandler())
+        .to("{{camel.route.emit-message}}");
+  }
+
+  void emitMessage() {
+    from("{{camel.route.emit-message}}")
+      .routeId("emit-msg")
+      .errorHandler(noErrorHandler())
+      .log("processing msg: ${body.payload}")
+      .setProperty("outboxMessage", simple("${body}"))
+      .choice()
+        // outbox_message has a not null constraint on message_routing_rule_id, so it is always
+        // present
+        // hence a check for null is not required
+        .when(simple("${body.messageRoutingRule.loginType} == 'OIDC'"))
+          .setProperty("credentials", simple("${body.messageRoutingRule.loginInformation}"))
+          .to("{{camel.route.oidc}}")
+        .otherwise()
+          .log("Unhandled loginType detected ${body.messageRoutingRule.loginType}")
+          .stop()
+      .end()
+      .to("{{camel.route.timestamp-notify}}");
+  }
+
+  void tokenOIDC() {
+    from("{{camel.route.oidc}}")
+      .routeId("oidc")
+      .onException(HttpOperationFailedException.class)
+        .continued(true)
+        .maximumRedeliveries("{{camel.max-redeliveries}}")
+        .redeliveryDelay("{{camel.redelivery-delay}}")
+        .useExponentialBackOff()
+        .backOffMultiplier(2)
+        .log(LoggingLevel.WARN, "Auth token request failed!")
+        .process(ex -> {
+          HashMap<String,String> tokenMap = new HashMap<>();
+          tokenMap.put("access_token", "<token>");
+          ex.getIn().setBody(tokenMap);
+        })
+        .marshal()
+        .json(JsonLibrary.Jackson)
+      .end()
+      .setHeader("CamelHttpMethod", constant("POST"))
+      .setHeader("Content-Type", simple("application/x-www-form-urlencoded"))
+      .setBody(
+          simple(
+              "grant_type=client_credentials&client_id=${exchangeProperty.credentials.clientID}&client_secret=${exchangeProperty.credentials.clientSecret}"))
+      .toD("${exchangeProperty.credentials.tokenURL}")
+      .log("auth response : ${body}")
+      .unmarshal()
+      .allowNullBody()
+      .json(JsonLibrary.Jackson)
+      .transform(simple("Bearer ${body[access_token]}"));
+  }
+
+  void timestampNotification() {
+    from("{{camel.route.timestamp-notify}}")
+      .routeId("timestamp-notify")
+      .onException(HttpOperationFailedException.class)
+        .handled(true)
+        .maximumRedeliveries("{{camel.max-redeliveries}}")
+        .redeliveryDelay("{{camel.redelivery-delay}}")
+        .useExponentialBackOff()
+        .backOffMultiplier(2)
+        .log("Timestamp request failed!")
```

Comment:
replace with logger component to use the applications logger provider.
https://camel.apache.org/components/3.18.x/log-component.html

Reply:
I think not required as camel looks up the registry and picks up the existing instance of slf4j, if I have understood this correctly.
https://camel.apache.org/components/next/eips/log-eip.html#_using_custom_logger_from_the_registry
 Is this what you meant?